### ISSUE TITLE
Item.reportinfo is missing documentation

### ISSUE THREAD

**Issue by The-Compiler** [pytest-dev/pytest#6689]
In the documentation, we have an example for [working with non-python tests](https://docs.pytest.org/en/latest/example/nonpython.html). There, we have:

```python
class YamlItem(pytest.Item):

    ...

    def reportinfo(self):
        return self.fspath, 0, "usecase: {}".format(self.name)
```

My first question when seeing this is "what does that tuple I return mean". Unfortunately, the API references for [Item](https://docs.pytest.org/en/latest/reference.html#_pytest.nodes.Item) and [Node](https://docs.pytest.org/en/latest/reference.html#_pytest.nodes.Node) don't help. Even reading the source doesn't help much:

```python
    def reportinfo(self) -> Tuple[Union[py.path.local, str], Optional[int], str]:
        return self.fspath, None, ""
```

Finally, looking at the Python plugin in pytest, things get clearer:

```python
        return fspath, lineno, modpath
```

but this should really be documented somewhere. Also, it's unclear to me why the example above uses `0` rather than `None`.

**Comment by Zac-HD** [comment]
Seems like a perfect case for a `namedtuple` (in this case `typing.NamedTuple`) to me - backwards-compatible and much more descriptive!

**Comment by nicoddemus** [comment]
> Seems like a perfect case for a namedtuple (in this case typing.NamedTuple) to me - backwards-compatible and much more descriptive!

Definitely, but unfortunately doing so would be an API breakage because `reportinfo` can be overriden by subclasses in plugins 😕.

For that reason I think we should stick to document it properly only.

**Comment by RonnyPfannschmidt** [comment]
we can follow-up with documenting the return value in terms of it and then putting up a union for it to sort it out as a migration for consumers/implementers (aka if we need to cast a return value, warn)
